globals/project.go

```diff
@@ -15,6 +15,11 @@ const (
 	DefaultHostName          = "cnoe.localtest.me"
 )
 
+var (
+	ClusterProtocol string // http or https scheme
+	ClusterPort     string // 8443 or user's port defined when the cluster is created
+)
+
```

Comment:
Can we get away without using these? How come we can't use `GiteaBaseUrl` and `ArgocdBaseUrl` as-is?

Reply:
No as we need such global vars to generate the proper URLs of ArgoCD or Gitea. Such variables are collected when the cluster is created. Without such vars it will be needed to fetch them using the kubernetes CR. 

```
k get localbuilds.idpbuilder.cnoe.io/dummy -oyaml
apiVersion: idpbuilder.cnoe.io/v1alpha1
kind: Localbuild
metadata:
  name: dummy
spec:
  buildCustomization:
    host: cnoe.localtest.me
    ingressHost: cnoe.localtest.me
    port: "9443"
    protocol: https
    selfSignedCert: |
      -----BEGIN CERTIFICATE-----
      MIIBsDCCA...EVSNlep0EtTdKee
      Sg0OoQ==
      -----END CERTIFICATE-----
  packageConfigs:
    argoPackageConfigs:
      enabled: true
    embeddedArgoApplicationsPackageConfigs:
      enabled: true
```

Reply:
I see. Can we not get it from the CR then? If custom packages exist, then the localbuild object must also exist correct?

Reply:
> Can we not get it from the CR then? If custom packages exist, then the localbuild object must also exist correct?

We can get it using go code doing this "kubectl get cr/localbuilds.idpbuilder.cnoe.io" but this is a bit boiler plate as we got such values when we execute the command "idp create" ....

Reply:
It's one call to the API server yea? i think that's fine. In Go, there's a lot of boilerplate like this 😆